### PR TITLE
Feature users boards

### PR DESCRIPTION
**Pull request by melonmanchan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,9 +19,11 @@
     "markdown": "0.5.0",
     "nuka-carousel": "0.0.12",
     "page": "1.6.0",
+    "query-string": "^2.3.0",
     "react": "0.13.1",
     "react-autosize-textarea": "0.2.2",
     "react-tween-state": "0.0.4",
+    "react-window-mixins": "0.0.6",
```
Comment (Temetz) [inline package.json]: remove

File: src/scripts/actions/board.js [hunk 2]
```diff
@@ -127,6 +125,25 @@ export default flux.actionCreator({
...
 
 	/**
+	 * Sets the user either active or away in a board
+	 */
+	setUserBoardActivity(boardId, payload) {
+		let token = UserStore.getToken();
+
+		return api.setUserBoardActivity({
+			          token:    token,
+                      isPoll:   payload.isPoll,
```
Comment (Temetz) [inline src/scripts/actions/board.js:135]: Tää sisennys on aika kauhee... edelleen

File: src/scripts/actions/user.js [hunk 3]
```diff
@@ -57,6 +57,21 @@ export default flux.actionCreator({
 			});
 	},
 
+	giveBoardAccess(boardId, accessCode) {
+		let token = UserStore.getToken();
+
+		return api.giveBoardAccess({ token: token, id: {
+			                         board:    boardId,
+								     code: accessCode} })
```
Comment (Temetz) [inline src/scripts/actions/user.js]: tab tab

File: src/scripts/actions/user.js [hunk 4]
```diff
@@ -57,6 +57,21 @@ export default flux.actionCreator({
...
+				return Promise.resolve();
+			})
+			.catch((err) => {
+				BroadcastAction.add(err, Action.User.GiveBoardAccess);
+				return Promise.reject();
+			});
+	},
+
 	/**
 	 *
```
Comment (Temetz) [inline src/scripts/actions/user.js]: Nice empty comment block

File: src/scripts/app.js [hunk 5]
```diff
@@ -11,6 +11,14 @@ import User      from './models/user';
...
 import WorkspaceView  from './views/workspace';
 
+import LoginView      from './views/form';
+import RegisterView   from './views/form';
+import GuestLoginView from './views/form';
+import UserAccessView from './views/form';
+
+import UserAction     from './actions/user';
+
+import qs from 'query-string';
```
Comment (Temetz) [inline src/scripts/app.js]: indents

File: src/scripts/app.js [hunk 6]
```diff
@@ -95,6 +103,14 @@ UserStore.addChangeListener(() => {
 	}
 });
 
+function authenticate(ctx, next) {
+		if(ctx.token = localStorage.getItem('access_token')) {
+
```
Comment (Temetz) [inline src/scripts/app.js]: Empty line

File: src/scripts/components/dialog/board-members.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        return this.props.onDismiss();
+    },
+
+    render() {
+        let board = this.props.board;
+        let members = board.members.sort(function(x, y) {
+            return new Date(y.lastSeen) - new Date(x.lastSeen);
+        });
+
+        //members = board.members.toJS();
```
Comment (Temetz) [inline src/scripts/components/dialog/board-members.js]: Why so much commented code?

File: src/scripts/components/navigation.js [hunk 8]
```diff
@@ -5,7 +5,8 @@ import Action          from '../actions';
 import UserAction      from '../actions/user';
 import BroadcastAction from '../actions/broadcast';
 
-import Dropdown  from '../components/dropdown';
+import Dropdown     from '../components/dropdown';
+import MemberDialog from '../components/dialog/board-members';
 import UserVoice from '../components/user-voice';
```
Comment (Temetz) [inline src/scripts/components/navigation.js:11]: indents

File: src/scripts/models/board.js [hunk 9]
```diff
@@ -45,14 +46,24 @@ const Size = immutable.Record({
...
+	_id:      '',
+	isActive: null,
+	role:     '',
+	lastSeen: null,
+	user:     new User()
+});
+
 const Board = immutable.Record({
-	id:               '',
+	id:               '',
```
Comment (Temetz) [inline src/scripts/models/board.js]: space

File: src/scripts/stores/board.js [hunk 10]
```diff
@@ -59,7 +92,7 @@ export default flux.store({
 			if(has(boards, payload.board)) {
 				return boards = remove(boards, payload.board);
 			}
-			throw new Error('Board not found!');
+			//throw new Error('Board not found!');
```
Comment (Temetz) [inline src/scripts/stores/board.js]: Maybe this shouldn't be commented? Check please.

File: src/scripts/utils/api.js [hunk 11]
```diff
@@ -50,7 +69,7 @@ function login(opts = {}) {
             token: res.headers['x-access-token']
         }
     });
-}
+}*/
```
Comment (Temetz) [inline src/scripts/utils/api.js]: quite a lot of commented code here

File: src/scripts/utils/request.js [hunk 12]
```diff
@@ -20,7 +20,7 @@ function request(to, options = {}) {
 		let request = to(options.url)
 			.set('Accept',        'application/json')
 			.set('Content-Type',  'application/json')
-			.set('Authorization', `Bearer ${options.token}`);
+			.set('Authorization', options.auth || `Bearer ${options.token}`)
```
Comment (Temetz) [inline src/scripts/utils/request.js]: Missing ;

File: src/scripts/views/board.js [hunk 13]
```diff
@@ -1,5 +1,5 @@
-import page  from 'page';
-import React from 'react/addons';
+import page         from 'page';
+import React        from 'react/addons';
```
Comment (Temetz) [inline src/scripts/views/board.js]: Check indents

File: src/scripts/views/board.js [hunk 14]
```diff
@@ -23,6 +23,7 @@ import EditBoardDialog   from '../components/dialog/edit-board';
 import ExportBoardDialog from '../components/dialog/export-board.js';
 import ShareBoardDialog  from '../components/dialog/share-board';
 
+
```
Comment (Temetz) [inline src/scripts/views/board.js]: newline

File: src/scripts/views/board.js [hunk 15]
```diff
@@ -41,9 +42,7 @@ export default React.createClass({
 		}
 	},
 
-	mixins: [
-		listener(UserStore, BoardStore, SettingsStore)
-	],
+	mixins: [ listener(UserStore, BoardStore, SettingsStore)	],
```
Comment (Temetz) [inline src/scripts/views/board.js]: Just use old style like above.

File: src/scripts/views/board.js [hunk 16]
```diff
@@ -61,19 +60,36 @@ export default React.createClass({
...
 
+	componentWillMount() {
+		this.setUserActivity({isActive: true, isPoll: false});
+	},
+
 	componentDidMount() {
 		BoardAction.load(this.props.id);
 		document.addEventListener('touchmove', preventDefault);
+
+		// Poll server every 5 seconds to indicate we're still alive!
```
Comment (Temetz) [inline src/scripts/views/board.js]: Maybe every 10 seconds as the timeout is 10000

File: src/scripts/actions/board.js (hunk 2 above)
Comment (Temetz) [inline src/scripts/actions/board.js:135]: Lines 134-139 use mixed spaces and tabs.

File: src/scripts/actions/user.js [hunk 17]
```diff
@@ -57,6 +57,20 @@ export default flux.actionCreator({
 			});
 	},
 
+	giveBoardAccess(boardId, accessCode) {
+		let token = UserStore.getToken();
+		return api.giveBoardAccess({ token: token, id: {
+			                         board:    boardId,
+								     code: accessCode} })
```
Comment (Temetz) [inline src/scripts/actions/user.js]: Lines 63-64 mixed tabs and spaces

File: src/scripts/utils/api.js [hunk 18]
```diff
@@ -249,6 +265,25 @@ function deleteTicket(opts = {}) {
...
 
+function setUserBoardActivity(opts = {}) {
+	let options = {
+		url:   `${API_URL}/boards/${opts.id.board}/setactivity`,
+		token: opts.token,
+		payload: opts.payload
+	}
+
+    if(opts.isPoll) {
+
```
Comment (Temetz) [inline src/scripts/utils/api.js:276]: Extra linebreaks 276, 278, 280, 283

**Comment by Temetz** [comment]
Ok. Coolio.

**Comment by melonmanchan** [comment]
Nice!!! :new_moon_with_face:

**Pull request merged**